homeassistant/components/switchbot/cover.py

```diff
@@ -99,15 +105,20 @@ async def async_set_cover_position(self, **kwargs: Any) -> None:
         position = kwargs.get(ATTR_POSITION)
 
         _LOGGER.debug("Switchbot to move at %d %s", position, self._address)
+        if position is not None:
+            self._attr_is_opening = position > self._attr_current_cover_position
+            self._attr_is_closing = position < self._attr_current_cover_position
         self._last_run_success = bool(await self._device.set_position(position))
```

Comment:
```suggestion
        self._last_run_success = bool(await self._device.set_position(position))
        self._attr_is_closing = self._device.is_closing()
        self._attr_is_opening = self._device.is_opening()        
```

Reply:
Not sure this will work but it would be better to have the lib be the single source of truth here

Reply:
I agree. Is there a potential race condition here though?

Reply:
Guess my concern is even if it "works" in my testing, is there a potential race condition that would cause it to fail sometimes? Granted, it would get updated during the next coordinator update so it wouldn't be "broken" it would just be delayed.

Reply:
There is also the chance the command fails and we still flip the attr

Reply:
Nevermind, that's what I get for reading a github suggestion on my phone, I thought you were removing the await. Tested and working consistently. Will raise PR in a min.